src/main/java/com/minecolonies/coremod/client/gui/WindowTownHallColonyManage.java

```diff
@@ -0,0 +1,192 @@
+
+            if (existingColony != null || IColonyManager.getInstance().isTooCloseToColony(world, pos))
+            {
+                findPaneOfTypeByID(TEXT_FEEDBACK, Text.class).setTextContent(LanguageHandler.format("com.minecolonies.coremod.gui.colony.denied.tooclose"));
+            }
+        }
+
+        if (MineColonies.getConfig().getCommon().restrictColonyPlacement.get())
+        {
+            final double spawnDistance = Math.sqrt(BlockPosUtil.getDistanceSquared2D(pos, world.getSpawnPoint()));
+            if (spawnDistance < MineColonies.getConfig().getCommon().minDistanceFromWorldSpawn.get())
+            {
+                findPaneOfTypeByID(TEXT_FEEDBACK, Text.class).setTextContent(LanguageHandler.format(CANT_PLACE_COLONY_TOO_CLOSE_TO_SPAWN,
+                  MineColonies.getConfig().getCommon().minDistanceFromWorldSpawn.get()));
+            }
+            else if (spawnDistance > MineColonies.getConfig().getCommon().maxDistanceFromWorldSpawn.get())
+            {
+                findPaneOfTypeByID(TEXT_FEEDBACK, Text.class).setTextContent(LanguageHandler.format(CANT_PLACE_COLONY_TOO_FAR_FROM_SPAWN,
+                  MineColonies.getConfig().getCommon().maxDistanceFromWorldSpawn.get()));
+            }
+        }
+
+        if (findPaneOfTypeByID(TEXT_FEEDBACK, Text.class).getTextContent().isEmpty())
+        {
+            findPaneOfTypeByID(BUTTON_CREATE, ButtonImage.class).enable();
+            findPaneOfTypeByID(TEXT_FEEDBACK, Text.class).setTextContent(LanguageHandler.format("com.minecolonies.coremod.gui.colony.allowed.create"));
+        }
+    }
+
+    @Override
+    public void onButtonClicked(@NotNull final Button button)
+    {
+        switch (button.getID())
+        {
+            case BUTTON_CLOSE:
+            {
+                close();
+                return;
+            }
+            case BUTTON_CREATE:
+            {
+                new VanillaParticleMessage(pos.getX(), pos.getY(), pos.getZ(), ParticleTypes.DRAGON_BREATH).onExecute(null, false);
+                Minecraft.getInstance().world.playSound(Minecraft.getInstance().player, Minecraft.getInstance().player.getPosition(),
+                  SoundEvents.BLOCK_CAMPFIRE_CRACKLE, SoundCategory.AMBIENT, 2.5f, 0.8f);
+                Network.getNetwork().sendToServer(new CreateColonyMessage(pos));
+                close();
+
+                return;
+            }
+            case BUTTON_DELETE:
+            {
+                new WindowTownHallColonyDelete().open();
+            }
+            default:
+        }
+    }
+
+    /**
+     * Finds the first nearby colony claim in the range
+     *
+     * @param world world to use
+     * @param start start position
+     * @param range search range
+     * @return
+     */
+    private static int findNextNearbyColony(final World world, final BlockPos start, final int range)
```

Comment:
On the client side?

Reply:
ye its for displaying nearby colonies, and the client doesnt have a full list of colonies, but does get claim data